### PR TITLE
SRVCOM-3319: Generate CSV with Konflux images (serving, eventing, operator)

### PR DESCRIPTION
**Pull request by pierDipi**

### CODE REVIEW

File: hack/lib/images.bash [hunk 1]
```diff
@@ -155,3 +155,21 @@ function knative_backstage_plugins_images_release() {
...
+  # digest=$(docker manifest inspect "${image_without_tag}:latest" -v | jq '.Descriptor.digest')
+  digest=''
+  if [ "${digest}" = "" ]; then
+    # exit 1
+    echo "${image_without_tag}:latest"
+    return
+  fi
+
+  echo "${image_without_tag}@${digest}"
+}
```
Comment (pierDipi) [inline hack/lib/images.bash:175]: Fix `TODO`

File: hack/lib/catalogsource.bash [hunk 2]
```diff
@@ -67,12 +72,13 @@ function install_catalogsource {
...
     # Replace bundle reference with previously built bundle
-    bundle="registry.ci.openshift.org/knative/release-${CURRENT_VERSION}:serverless-bundle"
+    bundle="${SERVERLESS_BUNDLE%:*}" # Remove the tag from the match
+    bundle="${SERVERLESS_BUNDLE%@*}" # Remove the sha from the match
     if ! grep "${bundle}" "${rootdir}/${index_dorkerfile_path}"; then
       logger.error "Bundle ${bundle} not found in Dockerfile."
       return 1
     fi
-    sed -i "s_\(.*\)\(${bundle}\)\(.*\)_\1image-registry.openshift-image-registry.svc:5000/$OLM_NAMESPACE/serverless-bundle:latest\3_" "${rootdir}/${index_dorkerfile_path}"
+    sed -ri "s#(.*)(${bundle})(:[a-z0-9]*)?(@sha[0-9]+:[a-z0-9]+)?(.*)#\1image-registry.openshift-image-registry.svc:5000/${OLM_NAMESPACE}/serverless-bundle:latest\5#" "${rootdir}/${index_dorkerfile_path}"
```
Comment (pierDipi) [inline hack/lib/catalogsource.bash:81]: Tested with:

```shell
$ bundle="quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-release-134/serverless-bundle"
$ sed -r "s#(.*)(${bundle})(:[a-z0-9]*)?(@sha[0-9]+:[a-z0-9]+)?(.*)#\1image-registry.openshift-image-registry.svc:5000/<namespace>/serverless-bundle:latest\5#" olm-catalog/serverless-operator/index/Dockerfile
FROM registry.ci.openshift.org/origin/4.15:operator-registry AS opm

FROM registry.access.redhat.com/ubi9/ubi-minimal as builder

COPY --from=opm /bin/opm /bin/opm

COPY olm-catalog/serverless-operator/index/configs /configs

RUN /bin/opm init serverless-operator --default-channel=stable --output yaml >> /configs/index.yaml
RUN /bin/opm render --skip-tls-verify -o yaml \
quay.io/openshift-knative/serverless-bundle:release-1.32.0 \
registry.ci.openshift.org/knative/release-1.33.0:serverless-bundle \
      registry.ci.openshift.org/knative/release-1.34.0:serverless-bundle >> /configs/index.yaml || \
    /bin/opm render --skip-tls-verify -o yaml \
quay.io/openshift-knative/serverless-bundle:release-1.32.0 \
registry.ci.openshift.org/knative/release-1.33.0:serverless-bundle \
      image-registry.openshift-image-registry.svc:5000/<namespace>/serverless-bundle:latest >> /configs/index.yaml

FROM registry.ci.openshift.org/origin/4.15:operator-registry

COPY --from=builder /configs /configs

LABEL operators.operatorframework.io.index.configs.v1=/configs

ENTRYPOINT ["/bin/opm"]
CMD ["serve", "/configs"]
```

Comment (pierDipi) [reply hack/lib/catalogsource.bash:81]: The idea is for matching:
1. `(.*)` match starting spaces and anything else until the next match
1. `(${bundle})` match the bundle image without tag or sha
1. `(:[a-z0-9]*)?`  optionally match tag
1. `(@sha[0-9]+:[a-z0-9]+)?`  optionally match sha
1. `(.*)` match the rest

Replace:

1. `\1` Use everything matched by 1
2. `image-registry.openshift-image-registry.svc:5000/${OLM_NAMESPACE}/serverless-bundle:latest` replace the image
3. `\5` use everything matched by 5

Comment (pierDipi) [reply hack/lib/catalogsource.bash:81]: image

File: hack/lib/images.bash [hunk 3]
```diff
@@ -3,16 +3,25 @@
...
-registry="${registry_host}/openshift"
+ci_registry_host='registry.ci.openshift.org'
+ci_registry="${ci_registry_host}/openshift"
+
 export CURRENT_VERSION_IMAGES=${CURRENT_VERSION_IMAGES:-"main"}
+CURRENT_VERSION="$(metadata.get project.version)"
+
+quay_registry_app_version=${CURRENT_VERSION/./} # 1.34.0 -> 134.0
+quay_registry_app_version=${quay_registry_app_version%.*} # 134.0 -> 134
+registry_host="quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-release-${quay_registry_app_version}"
```
Comment (pierDipi) [inline hack/lib/images.bash:14]: The current issue with this is that we wouldn't respect the per-component configuration

Comment (pierDipi) [reply hack/lib/images.bash]: Resolved in the last commit

**Review by creydr — COMMENTED** [review]
```
$ k -n knative-eventing get po eventing-controller-5dcb8787-4d4jb -ojsonpath='{.spec.containers[*].image}'
quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kn-eventing-controller@sha256:221b032b42a451896b2f73530b327934c8d662a67a6ea3b1e1e8bcfecd1a666f registry.ci.openshift.org/origin/4.15:kube-rbac-proxy
```

/lgtm

**Comment by pierDipi** [comment]
```
FAIL test/upgrade.TestServerlessUpgradePrePost/Run/Steps/PostUpgradeTests/Verify_jobs_in_knative-serving (-1.00s)

=== Failed
=== FAIL: test/upgrade  (0.00s)
panic: test timed out after 1h30m0s
running tests:
	TestServerlessUpgradePrePost (1h30m0s)
	TestServerlessUpgradePrePost/Run/Steps (1h27m38s)
	TestServerlessUpgradePrePost/Run/Steps/PostUpgradeTests (1h26m38s)
	TestServerlessUpgradePrePost/Run/Steps/PostUpgradeTests/Verify_jobs_in_knative-serving (1h26m38s)

goroutine 2732 [running]:
testing.(*M).startAlarm.func1()
	/usr/lib/golang/src/testing/testing.go:2366 +0x265
created by time.goFunc
	/usr/lib/golang/src/time/sleep.go:177 +0x45
```

**Comment by pierDipi** [comment]
It doesn't seem related

**Pull request merged**